src/System/Random/Primitive.agda

```diff
@@ -36,7 +36,7 @@ postulate
 {-# COMPILE GHC randomRIO-Int = randomRIO #-}
 {-# COMPILE GHC randomIO-Float = randomIO #-}
 {-# COMPILE GHC randomRIO-Float = randomRIO #-}
-{-# COMPILE GHC randomIO-Nat = randomIO #-}
+{-# COMPILE GHC randomIO-Nat = abs <$> randomIO #-}
```

Comment:
Nasty little bug here: `Nat` is `Integer` at runtime and so we need to be more careful here!

Reply:
If there ever were an ad for contributing a case to the testsuite when we mess with the FFI, that's it.